pageserver/src/tenant/storage_layer.rs

```diff
@@ -81,6 +88,103 @@ pub enum ValueReconstructResult {
     Missing,
 }
 
+#[derive(Default)]
+pub struct LayerAccessStats(RwLock<LayerAccessStatsInner>);
```

Comment:
As discussed, just use `Mutex` here.

Rationale:
- Write is used for competing accesses
- Read blocks writes, used for reading it

Outcome: We don't get any benefit from using a mutex as all access is blocking every other kind.